dial.go

```diff
@@ -0,0 +1,143 @@
+package tlsdialer
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"net"
+	"os"
+	"strings"
+
+	"github.com/tarantool/go-openssl"
+)
+
+func sslDialContext(ctx context.Context, network, address string,
+	sslOpts opts) (connection net.Conn, err error) {
+	var sslCtx interface{}
+	if sslCtx, err = sslCreateContext(sslOpts); err != nil {
+		return
+	}
+
+	return openssl.DialContext(ctx, network, address, sslCtx.(*openssl.Ctx), 0)
+}
+
+// interface{} is a hack. It helps to avoid dependency of go-openssl in build
+// of tests with the tag 'go_tarantool_ssl_disable'.
+func sslCreateContext(sslOpts opts) (ctx interface{}, err error) {
```

Comment:
We don't need the hack now.
```suggestion
func sslCreateContext(sslOpts opts) (sslCtx *openssl.Ctx}, err error) {
```